docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+# JavaScript Build Setup
+
+This page covers how to setup your development environment to use the ESNext syntax. This documentation covers development for your plugin to work with Gutenberg, to setup a development environment to work directly on Gutenberg, see the [CONTRIBUTING.md](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md) documentation.
+
+For simpler code, you can write JavaScript using ESNext and JSX syntax. This requires a transformation step to convert that syntax which a web browser may not understand, to one that it will.  Webpack and babel are the tools that perform this transformation step.
+
+[Webpack](https://webpack.js.org/) is a pluggable tool that processes JavaScript creating a compiled bundle that runs in a browser. [Babel](https://babeljs.io/) transforms JavaScript from one format to another. You use Babel as a plugin to Webpack to transform both ESNext and JSX to JavaScript.
+
+## Quick Start
+
+For a quick start, you can use one of the examples from the [Gutenberg Examples repository](https://github.com/wordpress/gutenberg-examples/). Each one of the `-esnext` directories contain the necessary files for working with ESNext and JSX.
+
+## Setup
+
+Both Webpack and Babel are tools written in JavaScript and run using [Node.js](https://nodejs.org/). Node.js is a runtime environment for JavaScript outside of a browser. Simply put, node allows you to run JavaScript code on the command-line.
+
+First step is setting up node for your development environment. This is dependent on your operating system, there are node packages available for most package systems. If you are using one of the package systems, setup can be as straight-forward as:
+```
+macOS: brew install node
+Windows: choco install node
+Ubuntu: apt install node
+```
```

Comment:
Just to avoid confusion about the commands, maybe format like:

```md
+ macOS: `brew install node`
+ Windows: `choco install node`
+ Ubuntu: `apt install node`
```

Reply:
(This separates the command from the OS)

Reply:
:+1: